Move NewModuleTest and NewCriterionTest from test_nn.py to common_nn.py

yf225 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35190 Refactor C++ API parity test mechanism and turn it on in CI again
* **#35189 Move NewModuleTest and NewCriterionTest from test_nn.py to common_nn.py**

Differential Revision: [D20588197](https://our.internmc.facebook.com/intern/diff/D20588197)

Commit:
Move NewModuleTest and NewCriterionTest from test_nn.py to common_nn.py

Commit:
Update on "Move NewModuleTest and NewCriterionTest from test_nn.py to common_nn.py"

Move NewModuleTest and NewCriterionTest from test_nn.py to common_nn.py

gh-metadata: pytorch pytorch 35189 gh/yf225/105/head

Commit:
Update on "Move NewModuleTest and NewCriterionTest from test_nn.py to common_nn.py"

Move NewModuleTest and NewCriterionTest from test_nn.py to common_nn.py

gh-metadata: pytorch pytorch 35189 gh/yf225/105/head

zou3519 reviewed (approved):
looks fine but the lint failures look legit

Commit:
Update on "Move NewModuleTest and NewCriterionTest from test_nn.py to common_nn.py"

Move NewModuleTest and NewCriterionTest from test_nn.py to common_nn.py

gh-metadata: pytorch pytorch 35189 gh/yf225/105/head
